### ISSUE TITLE
whatsapp-protocol.cc:105: error: expected primary-expression before '[' token

### ISSUE THREAD

**Issue by robert-scheck** [davidgfnet/whatsapp-purple#257]
$ g++ -c  -O2 -Wall -Wno-unused-function -fPIC -DPURPLE_PLUGINS -DPIC -I/usr/include/libpurple -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include   -std=c++0x -o whatsapp-protocol.o whatsapp-protocol.cc
    whatsapp-protocol.cc: In constructor 'WhatsappConnection::WhatsappConnection(std::string, std::string, std::string)':
    whatsapp-protocol.cc:105: error: expected primary-expression before '[' token
    whatsapp-protocol.cc:105: error: expected primary-expression before ']' token
    whatsapp-protocol.cc:105: error: expected primary-expression before 'char'
    $

This issue might be related to #256, but I would highly appreciate c++0x compatible code through.

**Comment by davidgfnet** [comment]
Yes, seems you got an old compiler. But AFAIK gcc 4.6 should be able to understand lambdas, right?

**Comment by robert-scheck** [comment]
Yes, this is GCC 4.4 from RHEL/CentOS 6. Any chance to avoid lambdas for at least some more time?
